libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -2010,9 +2011,33 @@ void ExpressionCompiler::appendExternalFunctionCall(
 
 	utils().popStackSlots(remainsSize);
 
-	if (returnSuccessCondition)
+	if (returnSuccessConditionAndReturndata)
 	{
-		// already there
+		// success condition is already there
+		// The return parameter types can be empty, when this function is used as
+		// an internal helper function e.g. for ``send`` and ``transfer``. In that
+		// case we're only interested in the success condition, not the return data.
+		if (!_functionType.returnParameterTypes().empty())
+		{
+			if (haveReturndatacopy)
+			{
+				solAssert(haveReturndatacopy, "");
+				m_context << Instruction::RETURNDATASIZE;
+				m_context.appendInlineAssembly(R"({
+                    switch v case 0 {
+                      v := 0x60
+                    } default {
+                        let size := v
+                        v := mload(0x40)
+				        mstore(0x40, add(v, and(add(size, 0x3f), not(0x1f))))
+                        mstore(v, size)
+				        returndatacopy(add(v, 0x20), 0, size)
+                    }
+			    })", {"v"});
+			}
+			else
+				m_context << u256(0x60);
```

Comment:
Ok - in ``docs/miscellaneous.rst`` under "Layout in Memory" it is just documented as "zero slot", so I thought it'd be fine to use it as empty memory array - but I changed it to create a "new" empty array now.

Reply:
Just out of curiosity: where is it actually used? I don't see any occurrence of ``0x60`` or ``96`` like that anywhere in a quick search through the code, nor did I find a constant or something similar with such values.

In any case: if it's not meant to be used like this, we should probably adjust ``docs/miscellaneous.rst``.

Reply:
I reverted the change again in the last commit, since at the moment it seems to me, that saving a memory slot by using ``zeroPointer`` is actually a good thing here. If there's a good reason not to, that I just don't see at the moment, we can just undo that commit.

Reply:
Ah ok - I'll look into InaccessibleDynamicType on pre-byzantium at some point then.